backend/tests/integration/records/serializers/json/test_authors.py

```diff
@@ -317,3 +317,25 @@ def test_authors_search_json_with_logged_in_cataloger(inspire_app):
 
     assert expected_status_code == response_status_code
     assert expected_result == response_data_hits_metadata
+
+
+def test_authors_detail_links(inspire_app):
+    expected_status_code = 200
+    record = create_record("aut")
+    expected_links = {
+        "json": f"http://localhost:5000/authors/{record['control_number']}?format=json"
```

Comment:
* Shouldn't this contain a `/api/` at the beginning? same question for all examples. I guess it's because you're using the  `inspire_app.test_client` and not the `test_client` of the `api` Flask application.
* Why isn't there a `self` link here? (again, same question for all record types)

Reply:
In test we can add /api/ or ignore it, then it will produce links accordingly. I can add it here but all requests in this test file are made without /api/ so I tired to be consistent with previous tests.

Reply:
self link in requirements was mentioned only on search result response not on record detail or search record details. If it should be also on record detail please update requirements ;)

Reply:
I think it makes sense also for the detailed records for consistency. What do you think?

Reply:
Please note that `self` will be actually the same link as `json` and it will add additional complexity to links generator to not add this self link in search record detail view.

Reply:
All I'm saying is we should do `self` everywhere or nowhere. I don't think that's the case currently.

Reply:
currently we have self only on `search result page` not on `search result item` nor on `detail record view`. Adding it to `detail view` will automatically add it to `search result item`, so there should be added additional check to not add them there as this will be messy if every record on `search page` will also have `self`. 
Additionally `self` is the same as `json` link, so maybe if we really want `self` everywhere we can replace `json` to `self`, but this looks messy IMHO.